Docs: Documents caching encryption in enterprise config (#36766)
grafana/grafana: docs/sources/enterprise/enterprise-configuration.md

The default is `1`.

## [caching.encryption]
### enabled

> Setting 'enabled' to `true` allows users to encrypt the query values in the cache.

Reviewer: ```suggestion
When 'enabled' to `true`, this setting allows users to encrypt the query values in the cache. Default is `false`.
```

Author: I updated it to
```
When 'enabled' is `true`, query values in the cache are encrypted.

The default is `false`.
```

Hopefully this is a bit more clear, but let me know what you think

> This value is `false` by default.

Reviewer: remove

Author: I updated it to be "The default is `false`.", but kept it on a different line to follow the pattern of the other settings in this markdown. Does that seem okay?

### encryption_key

> This string generates a key for encrypting the cache. For the encrypted cache data to persist between Grafana restarts, this key must be specified. If it is left empty when encryption is enabled, then the key is automatically generated on startup and the cache is cleared upon restarts.

Reviewer: ```suggestion
A string to generate a key for encrypting the cache. For the encrypted cache data to persist between Grafana restarts, you must specify this key. If it is empty when encryption is enabled, then the key is automatically generated on startup, and the cache clears upon restarts.
```

> The string used to generate a key for encrypting the cache. In order for encrypted cache data to persist between Grafana restarts, the key must be specified. If left empty when encryption is enabled, the key will be automatically generated on startup and the cache will be cleared upon restarts.

Reviewer: ```suggestion
This string generates a key for encrypting the cache. For the encrypted cache data to persist between Grafana restarts, this key must be specified. If it is left empty when encryption is enabled, then the key is automatically generated on startup and the cache is cleared upon restarts.
```

Author: Updated! Sorry for the delayed update - I was out all last week.
